examples/web-wagmi/src/profiles/UseFollow.tsx

```diff
@@ -0,0 +1,60 @@
+import { ProfileFieldsFragment, useExploreProfiles, useFollow } from '@lens-protocol/react';
+
+import { LoginButton } from '../components/auth/LoginButton';
+import { WhenLoggedIn, WhenLoggedOut } from '../components/auth/auth';
+import { Loading } from '../components/loading/Loading';
+import { ProfileCard } from './components/ProfileCard';
+
+type ProfileFollowProps = {
+  profile: ProfileFieldsFragment;
+};
+
+function FollowButton({ profile }: ProfileFollowProps) {
+  const { follow, isPending } = useFollow({ profile });
+
+  if (profile.isFollowedByMe || profile.isOptimisticFollowedByMe) return <p>Following</p>;
+
+  return (
+    <button onClick={follow} disabled={isPending}>
+      {isPending ? 'Following...' : 'Follow'}
```

Comment:
that state happens after a user clicks a Follow button and before the tx is confirmed in user's wallet. Some types of following require user submitting a tx.

Reply:
oh, good point, forgot about that part of UI

Reply:
Also in general we will need to provide the "ultimate fallback" for each tx where the tx is submitted and gas paid via the user's wallet. This happens only in case the API refuses to subsidize the tx in case some safety limit is reached.